src/main/groovy/org/embulk/plugins/gradle/tasks/MavenUploadEmbulkPluginJar.groovy

```diff
@@ -0,0 +1,60 @@
+package org.embulk.plugins.gradle.tasks
+
+import groovy.lang.Closure;
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.Upload
+
+/**
+ * MavenUploadEmbulkPluginJar is a variety of Upload tasks to upload Embulk's plugin JAR file to a Maven repository.
+ *
+ * It expects only Maven repositories because Embulk expects Maven repositories in its plugin system.
+ * It directly takes a "mavenDeployer" block, but its "pom" is eventually overridden in the task.
+ */
+class MavenUploadEmbulkPluginJar extends Upload {
+    @TaskAction
+    @Override
+    void upload() {
+        repositories {
+            mavenDeployer {
+                final Closure specifiedMavenDeployer = getMavenDeployerClosure().clone()
+                if (specifiedMavenDeployer != null) {
+                    specifiedMavenDeployer.setDelegate(owner)  // Set to run in the "mavenDeployer" block.
+                    specifiedMavenDeployer()
+                }
+
+                pom cleanupPomInPluginJar  // Override the pom
+            }
+        }
+        super.upload()
+    }
+
+    final Closure cleanupPomInPluginJar = {
+        // All dependencies except for "org.embulk:embulk-core" are removed.
+        whenConfigured { pom ->
+            pom.dependencies = pom.dependencies.findAll { dependency ->
+                ( dependency.groupId == "org.embulk" &&
+                  dependency.artifactId == "embulk-core" &&
+                  ( dependency.classifier == null || dependency.classifier == "" ) )
+            }
+        }
+
+        // The dependency "org.embulk:embulk-core" is "provided" for plugin JARs.
+        whenConfigured { pom ->
+            pom.dependencies.find { dependency ->
+                ( dependency.groupId == "org.embulk" &&
+                  dependency.artifactId == "embulk-core" &&
+                  ( dependency.classifier == null || dependency.classifier == "" ) )
+            }.scope = "provided"
+        }
+    }
```

Comment:
I'm not sure but, we need to care of `org.embulk:embulk-core` only? For example, how about `embulk-standards`?

Reply:
As this is building a plugin, I don't think it's normally depending only on `embulk-core`?

And it actually "works" even if it does have no any dependencies in its `pom`. I thought removing all dependencies was one option, but left `embulk-core` just in case...

Reply:
Understand. Sounds good for now.